corehq/apps/locations/forms.py

```diff
@@ -635,19 +635,32 @@ def clean_location_status_active(self):
             return False
         return None
 
+    def is_valid(self):
+        if not super().is_valid():
+            return False
+        location_id = self.cleaned_data.get('location_id')
+        if location_id is None:
+            return True
+        domain_membership = self.user.get_domain_membership(self.domain)
```

Comment:
Is there a reason not to use this function? That seems like the canonical way of doing this. 
 https://github.com/dimagi/commcare-hq/blob/daf905542b9f97d69acedc0a56101b82bf9a4d59/corehq/apps/locations/permissions.py#L270-L277